$OpenBSD: patch-jerror_c,v 1.4 2016/04/26 11:01:45 robert Exp $

This one is slightly problematic.  If an application allocates less
room for its error buffer than the recommended JMSG_LENGTH_MAX, the
error message buffer will still overflow.

--- jerror.c.orig	Mon Apr 25 11:01:31 2016
+++ jerror.c	Mon Apr 25 11:01:55 2016
@@ -188,9 +188,9 @@ format_message (j_common_ptr cinfo, char * buffer)
 
   /* Format the message into the passed buffer */
   if (isstring)
-    sprintf(buffer, msgtext, err->msg_parm.s);
+    snprintf(buffer, JMSG_LENGTH_MAX, msgtext, err->msg_parm.s);
   else
-    sprintf(buffer, msgtext,
+    snprintf(buffer, JMSG_LENGTH_MAX, msgtext,
             err->msg_parm.i[0], err->msg_parm.i[1],
             err->msg_parm.i[2], err->msg_parm.i[3],
             err->msg_parm.i[4], err->msg_parm.i[5],
